src/config/config_type.h

```diff
@@ -101,27 +103,30 @@ class MultiStringField : public ConfigField {
   std::vector<std::string> *receiver_;
 };
 
-class IntField : public ConfigField {
+template <typename IntegerType>
+class IntegerField : public ConfigField {
  public:
-  IntField(int *receiver, int n, int min, int max) : receiver_(receiver), min_(min), max_(max) { *receiver_ = n; }
-  ~IntField() override = default;
+  IntegerField(IntegerType *receiver, IntegerType n, IntegerType min, IntegerType max)
+      : receiver_(receiver), min_(min), max_(max) {
+    *receiver_ = n;
+  }
+  ~IntegerField() override = default;
   std::string ToString() override { return std::to_string(*receiver_); }
   Status ToNumber(int64_t *n) override {
     *n = *receiver_;
     return Status::OK();
   }
   Status Set(const std::string &v) override {
-    int64_t n;
-    auto s = Util::DecimalStringToNum(v, &n, min_, max_);
+    auto s = ParseInt(v, {min_, max_});
```

Comment:
```suggestion
    auto s = ParseInt<IntegerType>(v, {min_, max_});
```

Reply:
It can be deduced. It's better to explicitly pass type?

Reply:
I think it is better to pass type explicitly here.

Reply:
Done